src/components/FlavorStash/Note.js

```diff
@@ -0,0 +1,230 @@
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import React, { Component, Fragment } from 'react';
+import { bindActionCreators } from 'redux';
+import unified from 'unified';
+import parse from 'remark-parse';
+import remark2react from 'remark-react';
+import { Form as FinalForm, Field } from 'react-final-form';
+import { Button, ButtonGroup, Col, Form, InputGroup } from 'react-bootstrap';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { actions as noteActions } from 'reducers/note';
+import { getFlavorNote, isLoading } from 'selectors/note';
+
+export class Note extends Component {
+  static propTypes = {
+    actions: PropTypes.shape({
+      createNote: PropTypes.func.isRequired,
+      deleteNote: PropTypes.func.isRequired,
+      requestNote: PropTypes.func.isRequired,
+      updateNote: PropTypes.func.isRequired
+    }),
+    flavorId: PropTypes.string.isRequired,
+    loading: PropTypes.any.isRequired,
+    note: PropTypes.object.isRequired,
+    userId: PropTypes.string.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      editingNote: false,
+      updatedNote: null
+    };
+
+    this.handleNoteSubmit = this.handleNoteSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    const { actions, flavorId } = this.props;
+
+    actions.requestNote({ flavorId });
+  }
+
+  addNoteIcon(id) {
+    const { editingNote } = this.state;
+
+    return (
+      <Fragment>
+        {!editingNote && (
+          <Button
+            className="button-animation"
+            size="sm"
+            onClick={() => this.handleNoteEditor(id)}
+          >
+            <FontAwesomeIcon icon="plus" size="sm" title="Add Flavor Note" />
+          </Button>
+        )}
+      </Fragment>
+    );
+  }
+
+  editNoteIcon(id) {
+    const { editingNote } = this.state;
+
+    return (
+      <Fragment>
+        {!editingNote && (
+          <Button
+            className="button-animation"
+            size="sm"
+            onClick={() => this.handleNoteEditor(id)}
+          >
+            <FontAwesomeIcon icon="pen" size="sm" title="Edit Flavor Note" />
+          </Button>
+        )}
+      </Fragment>
+    );
+  }
+
+  handleNoteEditor() {
+    this.setState({ editingNote: !this.state.editingNote });
+  }
+
+  handleNoteSubmit(values) {
+    const { actions } = this.props;
+    const { note, update } = values;
+
+    if (!note) {
+      actions.deleteNote(values);
+    } else if (update === true) {
+      actions.updateNote(values);
+    } else {
+      actions.createNote(values);
+    }
+    this.setState({ updatedNote: note || '' });
```

Comment:
Why are we copying the note to local component state?

Reply:
It's mostly for user experience: it makes the update instant instead of having to pull it back from the API. But it also prevents the need for another GET request.

Reply:
I've refactored this, when you create or update a note it updates the collection in the reducer